pkg/block/block_linux.go

```diff
@@ -206,6 +206,23 @@ func diskPartitions(ctx *context.Context, paths *linuxpath.Paths, disk string) [
 		}
 		out = append(out, p)
 	}
+	// this is a disk with no partitions on it.
+	// Consider the filesystem on the disk to be a partition.
```

Comment:
codewise looks good. I'd like to have some (at least one) tests to cover this code before we can merge.

Reply:
Absolutely. However, I will admit after looking through the block tests, I'm not sure I understand the project's approach to tests. I don't see any mocking, so in order to run a full test suite someone would have to use physical hardware to replicate this? What are the chances someone will use a partition-less disk in testing?

Or are there mocks I'm missing and I'm just being a dunce 😆

Reply:
Hey @fromanirh can you give me a pointer to an example test to mimic? I'd like to wrap this up.

Reply:
@fromanirh hate to bother, but can I bump this? Would like to know what to add to get this merged.

Reply:
@taigrr sure thing, sorry for the dealy. On it.

Reply:
> Absolutely. However, I will admit after looking through the block tests, I'm not sure I understand the project's approach to tests. I don't see any mocking, so in order to run a full test suite someone would have to use physical hardware to replicate this? What are the chances someone will use a partition-less disk in testing?
> 
> Or are there mocks I'm missing and I'm just being a dunce laughing

Refactoring a bit the code to enable fake disks would work for me (cc @jaypipes though). The other option would be to actually capture the data reported by the system from a partitionless disk, using for example ghw-snapshot. A USB stick should be easy to set up and perfectly fit for the case.

Reply:
another option would be to have mock data under `testdata/'. This option sits between refactoring the code to enable unit tests and capturing real data with `ghw-snapshot`. I'd be fine with this approach as well

Reply:
Great, now I perfectly understand the testing setup. I'll get back on this soon.

Reply:
@fromanirh I've closed this PR in favor of a dedicated branch, Please see #308 which adds a unit test for this.